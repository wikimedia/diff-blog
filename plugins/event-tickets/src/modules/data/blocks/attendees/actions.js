/**
 * Internal dependencies
 */
import * as types from './types';

export const setTitle = ( title ) => ( {
	type: types.SET_ATTENDEES_TITLE,
	payload: {
		title,
	},
} );

export const setDisplayTitle = ( displayTitle ) => ( {
	type: types.SET_ATTENDEES_DISPLAY_TITLE,
	payload: {
		displayTitle,
	},
} );

export const setDisplaySubtitle = ( displaySubtitle ) => ( {
	type: types.SET_ATTENDEES_DISPLAY_SUBTITLE,
	payload: {
		displaySubtitle,
	},
} );

export const setInitialState = ( payload ) => ( {
	type: types.SET_ATTENDEES_INITIAL_STATE,
	payload,
} );
