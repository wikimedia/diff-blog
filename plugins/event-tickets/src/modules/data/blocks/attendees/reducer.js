/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	title: '',
	displayTitle: true,
	displaySubtitle: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_ATTENDEES_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_ATTENDEES_DISPLAY_TITLE:
			return {
				...state,
				displayTitle: action.payload.displayTitle,
			};
		case types.SET_ATTENDEES_DISPLAY_SUBTITLE:
			return {
				...state,
				displaySubtitle: action.payload.displaySubtitle,
			};
		default:
			return state;
	}
};
