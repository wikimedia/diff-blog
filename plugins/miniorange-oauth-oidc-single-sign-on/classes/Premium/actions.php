<?php


function show_extra_attributes($user)
{
    global $NQ;
    $SY = get_user_meta($user->ID, "\155\157\x5f\x6f\141\165\164\x68\x5f\143\x75\x73\x74\157\x6d\137\x61\x74\x74\162\151\142\165\x74\145\163");
    if (!$SY || !is_array($SY) || empty($SY)) {
        goto iu;
    }
    $SY = $SY[0];
    goto Tk;
    iu:
    return;
    Tk:
    ?>
	<h3>Extra profile information</h3>
	<table class="form-table" style="width:75%; border: 1px solid #aaa;">
		<tr>
			<td style="border: 1px solid #ccc;"><label for="user">User Name</label></td>
			<td style="border: 1px solid #ccc;"><strong><?php 
    echo esc_attr(get_the_author_meta("\144\x69\x73\x70\x6c\141\x79\137\x6e\141\155\x65", $user->ID));
    ?>
</strong></td>
		</tr>
		<?php 
    foreach ($SY as $ZZ => $Da) {
        ?>
			<tr>
				<td style="border: 1px solid #ccc;"><strong><?php 
        echo wp_kses($ZZ, get_valid_html());
        ?>
</strong></td>
				<td style="border: 1px solid #ccc;"><strong><?php 
        echo wp_kses($Da, get_valid_html());
        ?>
</strong></td>
			</tr>
		<?php 
        h9:
    }
    qi:
    ?>
	</table>
	<?php 
}
add_action("\x73\x68\157\x77\x5f\165\x73\145\x72\137\160\x72\157\x66\151\x6c\145", "\x73\150\x6f\167\x5f\145\x78\x74\x72\141\x5f\141\164\x74\162\151\142\x75\164\x65\x73");
add_action("\145\144\x69\x74\x5f\x75\x73\145\x72\x5f\x70\162\x6f\x66\151\x6c\x65", "\163\x68\x6f\167\137\x65\x78\x74\162\x61\137\141\164\164\x72\x69\x62\165\x74\x65\163");
function control_password_grant()
{
    global $NQ;
    $yZ = new MoOauthClient\GrantTypes\Password();
    $yZ->inject_ui();
    $yZ->inject_behaviour();
}
add_action("\x6d\x6f\137\x6f\141\x75\x74\x68\137\x63\x6c\151\x65\156\x74\x5f\141\144\144\137\160\x77\144\137\152\163", "\x63\157\156\x74\x72\x6f\x6c\x5f\x70\141\163\x73\167\x6f\162\x64\137\x67\x72\x61\x6e\x74");
function enqueue_pwd_essentials()
{
    ?>
	<link rel="stylesheet" href="<?php 
    echo MOC_URL . "\143\154\141\x73\163\145\163\57\120\162\145\x6d\151\x75\x6d\57\x72\x65\x73\x6f\x75\162\143\x65\163\57\x70\167\144\163\x74\x79\154\x65\56\143\x73\163";
    ?>
">
	<script src="<?php 
    echo MOC_URL . "\x63\154\141\163\163\145\x73\x2f\x50\162\145\155\151\x75\x6d\x2f\162\145\163\x6f\165\162\x63\x65\163\57\152\x71\x75\x65\x72\x79\56\155\151\x6e\56\x6a\x73";
    ?>
"></script>
	<script src="<?php 
    echo MOC_URL . "\x63\x6c\141\163\x73\x65\x73\57\x50\162\145\x6d\151\x75\155\57\162\x65\163\x6f\165\162\x63\145\163\x2f\x70\167\144\x2e\152\x73";
    ?>
"></script>
	<?php 
}
add_action("\160\x77\144\x5f\x65\x73\163\x65\x6e\x74\x69\x61\x6c\x73\137\151\x6e\164\x65\x72\x6e\x61\154", "\x65\x6e\161\165\145\x75\x65\x5f\x70\x77\144\x5f\145\x73\163\145\x6e\164\151\141\154\x73");
