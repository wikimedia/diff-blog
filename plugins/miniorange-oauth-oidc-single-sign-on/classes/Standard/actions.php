<?php


function mo_oauth_client_auto_redirect_external_after_logout()
{
    $NQ = new \MoOauthClient\Standard\MOUtils();
    $HC = $NQ->get_plugin_config();
    if (empty($HC->get_config("\x61\146\164\x65\162\137\x6c\x6f\147\x6f\x75\x74\137\165\x72\154"))) {
        goto YNg;
    }
    $o2 = $HC->get_config("\x61\146\x74\145\162\x5f\x6c\157\147\157\165\164\x5f\x75\x72\x6c");
    $ZN = get_current_user_id();
    $j6 = get_user_meta($ZN, "\x6d\x6f\137\157\x61\x75\x74\x68\x5f\x63\154\x69\145\156\x74\x5f\154\x61\163\164\x5f\x69\144\x5f\x74\x6f\x6b\x65\156", true);
    $o2 = str_replace("\43\x23\151\x64\137\164\x6f\153\x65\x6e\43\43", $j6, $o2);
    wp_redirect($o2);
    die;
    YNg:
}
add_action("\167\x70\137\x6c\x6f\147\x6f\165\164", "\x6d\x6f\x5f\x6f\141\x75\x74\x68\x5f\143\x6c\151\x65\x6e\x74\x5f\x61\165\x74\x6f\x5f\162\145\x64\151\162\145\x63\x74\x5f\145\170\164\145\162\x6e\x61\x6c\137\x61\x66\164\145\162\137\154\x6f\147\157\x75\x74");
