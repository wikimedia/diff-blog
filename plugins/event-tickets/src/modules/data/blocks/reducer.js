/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import rsvp from './rsvp';
import ticket from './ticket';
import attendees from './attendees';

export default combineReducers( {
	rsvp,
	ticket,
	attendees,
} );
