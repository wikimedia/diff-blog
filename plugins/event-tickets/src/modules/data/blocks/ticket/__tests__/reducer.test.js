/**
 * Internal dependencies
 */
import reducer, { DEFAULT_STATE } from '../reducer';
import * as actions from '../actions';

describe( 'Reducer', () => {
	it( 'should set the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'should set the is selected', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsSelected( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is settings open', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsSettingsOpen( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is settings loading', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsSettingsLoading( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the provider', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsProvider( 'provider' ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the shared capacity', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsSharedCapacity( 99 ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the temp shared capacity', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsTempSharedCapacity( 99 ),
		) ).toMatchSnapshot();
	} );
} );
