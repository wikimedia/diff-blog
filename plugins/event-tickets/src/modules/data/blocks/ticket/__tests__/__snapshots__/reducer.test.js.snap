// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer should set the is selected 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": true,
  "isSettingsLoading": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;

exports[`Reducer should set the is settings loading 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": false,
  "isSettingsLoading": true,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;

exports[`Reducer should set the is settings open 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": false,
  "isSettingsLoading": false,
  "isSettingsOpen": true,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;

exports[`Reducer should set the provider 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": false,
  "isSettingsLoading": false,
  "isSettingsOpen": false,
  "provider": "provider",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;

exports[`Reducer should set the shared capacity 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": false,
  "isSettingsLoading": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": 99,
  "tempSharedCapacity": "",
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;

exports[`Reducer should set the temp shared capacity 1`] = `
Object {
  "headerImage": Object {
    "alt": "",
    "id": 0,
    "src": "",
  },
  "isSelected": false,
  "isSettingsLoading": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": 99,
  "tickets": Object {
    "allClientIds": Array [],
    "byClientId": Object {},
  },
}
`;
